src/transformers/generation/flax_utils.py

```diff
@@ -458,9 +466,11 @@ def generate(
                 generation_config.max_length,
                 generation_config.pad_token_id,
                 generation_config.eos_token_id,
+                output_scores=generation_config.output_scores,
                 length_penalty=generation_config.length_penalty,
                 early_stopping=generation_config.early_stopping,
                 logits_processor=logits_processor,
+                num_beam_hyps_to_keep=generation_config.num_return_sequences,
```

Comment:
`num_return_sequences` would be the preferred argument name. Check the TF implementation, which is also XLA compatible!

(the same comment applies to other occurrences along the file :) )